File: snippets/go/lnurl_pay.go
package example

import (
	"log"

	"github.com/breez/breez-sdk-liquid-go/breez_sdk_liquid"
)

func PrepareLnurlPay(sdk *breez_sdk_liquid.BindingLiquidSdk) {
	// ANCHOR: prepare-lnurl-pay
	// Endpoint can also be of the form:
	// lnurlp://domain.com/lnurl-pay?key=val
	// lnurl1dp68gurn8ghj7mr0vdskc6r0wd6z7mrww4excttsv9un7um9wdekjmmw84jxywf5x43rvv35xgmr2enrxanr2cfcvsmnwe3jxcukvde48qukgdec89snwde3vfjxvepjxpjnjvtpxd3kvdnxx5crxwpjvyunsephsz36jf
	lnurlPayUrl := "lightning@address.com"

	if input, err := sdk.Parse(lnurlPayUrl); err != nil {
		switch inputType := input.(type) {
		case breez_sdk_liquid.InputTypeLnUrlPay:
			var amount breez_sdk_liquid.PayAmount = breez_sdk_liquid.PayAmountBitcoin{
				ReceiverAmountSat: uint64(5_000),
			}
			optionalComment := "<comment>"
			optionalValidateSuccessActionUrl := true

			req := breez_sdk_liquid.PrepareLnUrlPayRequest{
				Data:                     inputType.Data,
				Amount:                   amount,
				Comment:                  &optionalComment,
				ValidateSuccessActionUrl: &optionalValidateSuccessActionUrl,
			}
			prepareResponse, err := sdk.PrepareLnurlPay(req)
			if err != nil {
				log.Printf("Error: %#v", err)
				return
			}

			// If the fees are acceptable, continue to create the LNURL Pay
			feesSat := prepareResponse.FeesSat
			log.Printf("Fees: %v sats", feesSat)
		}
	}
	// ANCHOR_END: prepare-lnurl-pay
}

func PrepareLnurlPayDrain(sdk *breez_sdk_liquid.BindingLiquidSdk, data breez_sdk_liquid.LnUrlPayRequestData) {
	// ANCHOR: prepare-lnurl-pay-drain
	var amount breez_sdk_liquid.PayAmount = breez_sdk_liquid.PayAmountDrain{}
	optionalComment := "<comment>"
	optionalValidateSuccessActionUrl := true

	req := breez_sdk_liquid.PrepareLnUrlPayRequest{
		Data:                     data,
		Amount:                   amount,
		Comment:                  &optionalComment,
		ValidateSuccessActionUrl: &optionalValidateSuccessActionUrl,
	}
	prepareResponse, err := sdk.PrepareLnurlPay(req)
	if err != nil {
		log.Printf("Error: %#v", err)
		return
	}
	// ANCHOR_END: prepare-lnurl-pay-drain
	log.Printf("prepareResponse: %#v", prepareResponse)
}

func LnurlPay(sdk *breez_sdk_liquid.BindingLiquidSdk, prepareResponse breez_sdk_liquid.PrepareLnUrlPayResponse) {
	// ANCHOR: lnurl-pay
	req := breez_sdk_liquid.LnUrlPayRequest{
		PrepareResponse: prepareResponse,
	}
	if result, err := sdk.LnurlPay(req); err != nil {
		log.Printf("Result: %#v", result)
	}
	// ANCHOR_END: lnurl-pay
}

File: snippets/csharp/LnurlPay.cs
using Breez.Sdk.Liquid;

public class LnurlPaySnippets
{
    public void PrepareLnurlPay(BindingLiquidSdk sdk)
    {
        // ANCHOR: prepare-lnurl-pay
        // Endpoint can also be of the form:
        // lnurlp://domain.com/lnurl-pay?key=val
        // lnurl1dp68gurn8ghj7mr0vdskc6r0wd6z7mrww4excttsv9un7um9wdekjmmw84jxywf5x43rvv35xgmr2enrxanr2cfcvsmnwe3jxcukvde48qukgdec89snwde3vfjxvepjxpjnjvtpxd3kvdnxx5crxwpjvyunsephsz36jf
        var lnurlPayUrl = "lightning@address.com";

        try
        {
            var input = sdk.Parse(lnurlPayUrl);
            if (input is InputType.LnUrlPay lnurlp)
            {
                var amount = new PayAmount.Bitcoin(5000);
                var optionalComment = "<comment>";
                var optionalValidateSuccessActionUrl = true;
                
                var req = new PrepareLnUrlPayRequest(
                    lnurlp.data, 
                    amount, 
                    optionalComment, 
                    optionalValidateSuccessActionUrl);
                var prepareResponse = sdk.PrepareLnurlPay(req);

                // If the fees are acceptable, continue to create the LNURL Pay
                var feesSat = prepareResponse.feesSat;
                Console.WriteLine($"Fees: {feesSat} sats");
            }
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: prepare-lnurl-pay
    }
    public void PrepareLnurlPayDrain(BindingLiquidSdk sdk, LnUrlPayRequestData data)
    {
        // ANCHOR: prepare-lnurl-pay-drain
        try
        {
            var amount = new PayAmount.Drain();
            var optionalComment = "<comment>";
            var optionalValidateSuccessActionUrl = true;
            
            var req = new PrepareLnUrlPayRequest(
                data, 
                amount, 
                optionalComment, 
                optionalValidateSuccessActionUrl);
            var prepareResponse = sdk.PrepareLnurlPay(req);
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: prepare-lnurl-pay-drain
    }

    public void LnurlPay(BindingLiquidSdk sdk, PrepareLnUrlPayResponse prepareResponse)
    {
        // ANCHOR: lnurl-pay
        try
        {
            var result = sdk.LnurlPay(new LnUrlPayRequest(prepareResponse));
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: lnurl-pay
    }
}
